docs/contracts/events.rst

```diff
@@ -24,9 +24,10 @@ because the contract can only see the last 256 block hashes.
 
 You can add the attribute ``indexed`` to up to three parameters which adds them
 to a special data structure known as :ref:`"topics" <abi_events>` instead of
-the data part of the log. If you use arrays (including ``string`` and ``bytes``)
-as indexed arguments, its Keccak-256 hash is stored as a topic instead, this is
-because a topic can only hold a single word (32 bytes).
+the data part of the log.
+A topic can only hold a single word (32 bytes) so if you use a :ref:`reference type
+<reference-types>` for an indexed argument, Keccak-256 hash of the value is stored
```

Comment:
Perhaps we should list what reference types are supported? I think it would just be `arrays` and `structs`.

Reply:
One of the reasons for this PR is that I noticed that structs are not mentioned and I replaced the text to avoid having the same problem in the future when we add another reference type :)

But I can add that back if you think it was better.

Reply:
Okay, but we don't support all of them currently, right? For example `mapping`. 

Yeah, I'm fine with both. The user would get an error when they use `mapping` or other reference type anyway.

Reply:
Yeah, I think "reference type" is fine without listing all of them.

Reply:
> Okay, but we don't support all of them currently, right? For example mapping.

I think we do. It's surprising but mapping is not actually a reference type. It's not listed under "Reference Types" in the docs and also in the code `MappingType` does not inherit from `ReferenceType`.